jetty-util/src/main/java/org/eclipse/jetty/util/Pool.java

```diff
@@ -345,45 +417,48 @@ public void close()
 
     public int size()
     {
-        return sharedList.size();
+        return entries.size();
     }
 
     public Collection<Entry> values()
     {
-        return Collections.unmodifiableCollection(sharedList);
+        return Collections.unmodifiableCollection(entries);
     }
 
     @Override
     public void dump(Appendable out, String indent) throws IOException
     {
-        Dumpable.dumpObjects(out, indent, this);
+        Dumpable.dumpObjects(out, indent, this,
+            new DumpableCollection("entries", entries));
     }
 
     @Override
     public String toString()
     {
-        return String.format("%s@%x[size=%d closed=%s entries=%s]",
+        return String.format("%s@%x[size=%d closed=%s pending=%d]",
             getClass().getSimpleName(),
             hashCode(),
-            sharedList.size(),
+            entries.size(),
             closed,
-            sharedList);
+            pending.get());
     }
 
     public class Entry
     {
         // hi: positive=open/maxUsage counter; negative=closed; MIN_VALUE pending
         // lo: multiplexing counter
         private final AtomicBiInteger state;
+        private final int index;
```

Comment:
This index doesn't seem to be used.

Reply:
oops yes - that was a broken hack for another esoteric round robin